src/artifactContentFileSystemProvider.ts

```diff
@@ -0,0 +1,333 @@
+    public readDirectory(uri: vscode.Uri): [string, vscode.FileType][] {
+        return [];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public async readFile(uri: vscode.Uri): Promise<Uint8Array> {
+        const artifactContent: ArtifactInformationFromUri = await ArtifactContentFileSystemProvider.getArtifactInformationFromUri(uri);
+        const artifactContentRenderer: ArtifactContentRenderer | undefined = tryCreateRendererForArtifactContent(artifactContent.log, artifactContent.contents, artifactContent.parsedUriData.runIndex, artifactContent.parsedUriData.artifactIndex);
+
+        if (artifactContentRenderer) {
+            return Buffer.from(artifactContentRenderer.render(artifactContent.parsedUriData.artifactUri));
+        }
+
+        throw new Error(`There is no contents that can be rendered associated with artifact index ${artifactContent.parsedUriData.runIndex} for run index ${artifactContent.parsedUriData.runIndex}.`);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public createDirectory(uri: vscode.Uri): void {
+        throw new Error('Not implemented');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public writeFile(uri: vscode.Uri, content: Uint8Array, options: { create: boolean; overwrite: boolean }): void {
+        throw new Error('Not implemented');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public delete(uri: vscode.Uri, options: { recursive: boolean }): void {
+        throw new Error('Not implemented');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public rename(oldUri: vscode.Uri, newUri: vscode.Uri, options: { overwrite: boolean }): void {
+        throw new Error('Not implemented');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public dispose(): void {
+        vscode.Disposable.from(...this.disposables).dispose();
+        this.disposables = [];
+    }
+
+    /**
+     * Creates an embedded content URI.
+     * @param sarifLog The raw sarif log.
+     * @param logPath The full path to the SARIF log file.
+     * @param artifactUri The file name that VSCode will display in the editor and use for detection of type.
+     * @param runIndex The index of the run in the SARIF file.
+     * @param artifactIndex The artifact index.
+     * @param requiredExtension Indicates how to render binary content.
+     */
+    public static tryCreateUri(sarifLog: sarif.Log, logPath: vscode.Uri, artifactUri: vscode.Uri, runIndex: number, artifactIndex: number, requiredExtension: string | undefined): vscode.Uri | undefined {
+        if (!logPath.isSarifFile()) {
+            throw new Error(`${logPath.toString()} is not a SARIF file`);
+        }
+
+        const run: sarif.Run | undefined = sarifLog.runs[runIndex];
+        if (!run || !run.artifacts) {
+            return undefined;
+        }
+        const artifact: sarif.Artifact | undefined = run.artifacts[artifactIndex];
+        if (!artifact || !artifact.contents) {
+            return undefined;
+        }
+
+        let uriPath: string;
```

Comment:
Why is `uriPath` needed? It is ignored downstream, including by `parseUri()`.

Reply:
uriPath (which is the end of the URI before the path)  contains the file extension that VSCode will use to detect the file-type. I've added a comment.

---
In reply to: [411818093](https://github.com/microsoft/sarif-vscode-extension/pull/252#discussion_r411818093) [](ancestors = 411818093)